shared/js/content-scope/protections.js

```diff
@@ -0,0 +1,33 @@
+import { initStringExemptionLists, isFeatureBroken } from './utils'
+
+export async function initProtection (args) {
+    // don't inject into non-HTML documents (such as XML documents)
+    // but do inject into XHTML documents
+    if (document instanceof HTMLDocument === false && (
+        document instanceof XMLDocument === false ||
+        document.createElement('div') instanceof HTMLDivElement === false
+    )) {
+        return
+    }
+
+    const protections = [
+        'canvas',
+        'audio',
+        'temporary-storage',
+        'referrer',
+        'battery',
+        'screen-size',
+        'hardware',
+        'do-not-track',
+        'floc',
+        'gpc'
+    ]
+
+    initStringExemptionLists(args)
+    for (const protection of protections) {
+        const { init } = await import(`./${protection}-protection.js`)
+        if (!isFeatureBroken(args, protection)) {
+            init(args)
+        }
```

Comment:
If the import is `await`ed is could this cause a race condition? Or does this get made sync by the bundler?

Reply:
Assuming we were disabling on page load, there is one tick that each of these will cause. I think it's a risk we can take but I'm happy to add a comment for that effect when we come to that? The output code essentially does `Promise.resolve().then(code)`

Reply:
Can we do something like:
```js
protections.forEach((protection) => ...)
```
This would mean that all imports are requested on the first tick. With the async loop you currently have the last protection won't get imported until `protections.length` ticks.

Follow up, why let the import run at parse time (i.e. when we import this file)?